sbin/submitty_autograding_shipper.py

```diff
@@ -411,25 +420,38 @@ def unpack_job(which_machine, which_untrusted, next_directory, next_to_grade, ra
             "ERROR: the submitty_autograding_shipper.py script must be run by the DAEMON_USER"
         )
 
-    if which_machine == 'localhost':
-        address = which_machine
-    else:
-        address = which_machine.split('@')[1]
+    # Grab the path to this assignment for logging purposes
+    obj = packer_unpacker.load_queue_file_obj(JOB_ID, next_directory, next_to_grade)
+    if "generate_output" not in obj:
+        partial_path = os.path.join(obj["gradeable"], obj["who"], str(obj["version"]))
+        item_name = os.path.join(obj["semester"], obj["course"], "submissions", partial_path)
+    elif obj["generate_output"]:
+        item_name = os.path.join(obj["semester"], obj["course"], "generated_output")
+    is_batch = "regrade" in obj and obj["regrade"]
+
+    # Address is either localhost or a string of the form user@host
+    address = which_machine if which_machine == 'localhost' else which_machine.split('@')[1]
+
+    # The full name of the worker associated with the socket
+    worker_name = f"{socket.getfqdn()}_{address}_{which_untrusted}"
 
-    fully_qualified_domain_name = socket.getfqdn()
-    servername_workername = "{0}_{1}".format(fully_qualified_domain_name, address)
     target_results_zip = os.path.join(
         SUBMITTY_DATA_DIR, "autograding_DONE",
-        f"{servername_workername}_{which_untrusted}_results.zip"
+        f"{worker_name}_results.zip"
     )
     target_done_queue_file = os.path.join(
         SUBMITTY_DATA_DIR, "autograding_DONE",
-        f"{servername_workername}_{which_untrusted}_queue.json"
+        f"{worker_name}_queue.json"
     )
 
+    # status will be set to a GradingStatus.
+    status = None
+
+    # If the worker we are retrieving from is local
     if which_machine == "localhost":
```

Comment:
This comment as is is not very helpful to someone new to this code, as the if statement already covers that we're retrieving a localhost worker (or not). It would be more helpful to comment here instead why the differentiation is made between local vs remote for the steps that follow (if the code itself does not make that clear enough).

Reply:
Updated the comment.